### PR TITLE
Add is_annual_plan to DefaultProductPlan

### PR DESCRIPTION
**Pull request by nospame**
## Technical Summary
[SaaS-15492](https://dimagi.atlassian.net/browse/SAAS-15492)
First step to allowing self-service renewals of CommCare annual subscriptions.

## Feature Flag
Not feature flagged.

### Safety story
This code change and migration is small, affects only a handful of objects, and ensures a default (`False`) is added that is equivalent to existing behavior. Tested locally and on staging.

### Automated test coverage
`DefaultProductPlan` and its `get_default_plan_version` are used by several tests, showing that existing behavior is retained here. Changes in [SaaS-15493](https://dimagi.atlassian.net/browse/SAAS-15493) will include testing of this new property through `test_ensure_plans`.

### QA Plan
No QA planned for this particular change.

### Migrations

- [X] The migrations in this code can be safely applied first independently of the code

### Rollback instructions
- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by jingcheng16 — APPROVED** [review]
Looks good to to me! Thank you!

**Comment by nospame** [comment]
Holding off merging because https://github.com/dimagi/commcare-hq/pull/34515 also adds a 0092 migration in `accounting`. Will most likely update this to 0093 after that PR is merged.

**Review by jingcheng16 — APPROVED** [review]

**Pull request merged**
